rotkehlchen/globaldb/updates.py

```diff
@@ -399,22 +398,35 @@ def _process_multiasset_mapping(
             connection: 'DBConnection',
             action: str,
             full_insert: str,
+            version: int,
     ) -> None:
-        """
-        Process the insertion of a new asset_collection mapping
+        """Process the insertion of a new asset_collection mapping.
+
+        After version 30, multiasset_mappings table includes an 'is_main_asset' column to
+        identify the main token in collections. This changes both the table structure and
+        the expected number of matched groups in the regex pattern (3 groups before v30,
+        4 groups after).
+
         May raise:
         - DeserializationError
         - UnknownAsset
         """
-        mapping_match = self.multiasset_mappings_re.match(full_insert)
+        if version > 30:
```

Comment:
as discussed in the call let's get rid of the if/else check and let's enforce the non-backwards compatibility for assets updates.

But to not have users lose assets updates, as Yabir suggested, whenever a global DB upgrade happens for which the asset update will needs its min/max schema change, then force a remote asset update by automatically choosing "select all remote"